packages/actor-extract-links-extract-tree/lib/SolutionDomain.ts

```diff
@@ -0,0 +1,198 @@
+import { SolutionRange } from './SolutionRange';
+import { LogicOperator } from './solverInterfaces';
+
+const AN_EMPTY_SOLUTION_RANGE = new SolutionRange(undefined);
+/**
+ * A class representing the domain of a solution of system of boolean equation.
+ * Every operation return a new object.
+ */
+export class SolutionDomain {
+  /**
+     * The multiple segment of the domain, it is always order by the lower bound
+     * of the SolutionRange.
+     */
+  private domain: SolutionRange[] = [];
+
+  /**
+    * The last operation apply to the domain
+    *
+    */
+  private lastOperation: LogicOperator | undefined = undefined;
```

Comment:
I don't quite understand why operations are stored inside the domain.
This feels like a violation of the separation of concerns.